### PR TITLE
Savefile sub directories

### PR DESCRIPTION
**Pull request by abairmj**
Adds subdirectories to save files for issue #8

### CODE REVIEW

File: test_host.cpp [hunk 1]
```diff
@@ -204,13 +204,34 @@ void TestHost::DrawVertices(uint32_t elements) {
...
+  }
+  if (!CreateDirectory(folder_path.c_str(), nullptr) && GetLastError() != ERROR_ALREADY_EXISTS) {
     assert(!"Failed to create output directory.");
   }
+}
+
+// Returns the full output filepath including the filename
+// Creates output directory if it does not exist
+std::string TestHost::PrepareSaveFilePNG(std::string output_directory, const std::string &filename) {
+  std::replace(output_directory.begin(), output_directory.end(), ' ', '_');
```
Comment (abaire) [inline test_host.cpp:219]: Nit: since we know we're already doing this cleanup in the constructor, we can probably omit it here. It also may be surprising behavior if the caller expected the output dir to have spaces in it for whatever reason.

File: test_host.cpp [hunk 2]
```diff
@@ -204,13 +204,34 @@ void TestHost::DrawVertices(uint32_t elements) {
...
+  EnsureFolderExists(output_directory);
+
+  output_directory += "\\";
+  output_directory += filename;
+  output_directory += ".png";
+
+  if (output_directory.length() > MAX_FILE_PATH_SIZE) {
+    assert(!"Full Save File Path is too long.");
```
Comment (abaire) [inline test_host.cpp:227]: Nit: lcase "Save File Path"?

File: test_host.h [hunk 3]
```diff
@@ -82,8 +83,10 @@ class TestHost {
  private:
   void SetupControl0() const;
   void SetupTextureStages() const;
-  static void SaveBackbuffer(const char *output_directory, const char *name);
-  void SaveZBuffer(const char *output_directory, const char *name) const;
+  static void EnsureFolderExists(const std::string &folder_path);
+  static std::string PrepareSaveFilePNG(std::string output_directory, const std::string &filename);
+  static void SaveBackBuffer(const std::string &output_directory, const std::string &name);
+  void SaveZBuffer(const std::string &output_directory, const std::string &name) const;
```
Comment (abaire) [inline test_host.h:89]: Nit (and not something you introduced) - can this be static?

File: tests/two_d_line_tests.cpp [hunk 4]
```diff
@@ -106,7 +106,7 @@ static std::string ColorFormatName(uint32_t format, bool ReturnShortName) {
       if (ReturnShortName) return "15";
       return "R5G5B5";
     case NV05C_SET_COLOR_FORMAT_LE_X8R8G8B8:
-      if (ReturnShortName) return "32";
+      if (ReturnShortName) return "24";
```
Comment (abaire) [inline tests/two_d_line_tests.cpp:109]: Super nit (ignore it for this PR) - this change doesn't really have anything to do with the other fixes, so ideally it'd be a separate PR.

**Review by abaire — APPROVED** [review]
Few nitpicks. You can merge as is or fix up if you feel like it.

File: test_host.cpp (hunk 1 above)
Comment (abairmj) [reply test_host.cpp]: done

File: test_host.cpp (hunk 2 above)
Comment (abairmj) [reply test_host.cpp]: done

File: test_host.h (hunk 3 above)
Comment (abairmj) [reply test_host.h:89]: uses non static member vars, left as is

File: tests/two_d_line_tests.cpp (hunk 4 above)
Comment (abairmj) [reply tests/two_d_line_tests.cpp:109]: next time!
